File: frontend/src/types/incident.ts
/**
 * TypeScript mirror of the Incident API schemas (Phase 1 Step 7.3).
 */

export type IncidentStatus =
  | 'open'
  | 'in_progress'
  | 'resolved'
  | 'false_positive'
  | 'closed'

export interface Incident {
  id: string
  alert_group_id: string
  title: string
  description: string | null
  severity: string
  /** Snapshot copied from EventRisk when the case was created. */
  risk_score: number
  status: IncidentStatus
  disposition: string | null
  created_at: string
  updated_at: string
  resolved_at: string | null
  closed_at: string | null
}

export interface IncidentListResponse {
  total: number
  page: number
  size: number
  items: Incident[]
}

File: backend/app/schemas/incident.py
"""Pydantic schemas of the Incident Management API."""
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class IncidentCreate(BaseModel):
    """POST /incidents — only the event is given; the case record
(title/severity/description/risk_score) is auto-filled by the service."""

    alert_group_id: uuid.UUID


class IncidentStatusUpdate(BaseModel):
    """PATCH /incidents/{id}/status — an explicit action route rather than a
generic field update: ``status`` is a requested transition, checked
against the service-layer state machine, so an unknown vocabulary value
answers 409 instead of a 422 body-validation error."""

    status: str


class IncidentRead(BaseModel):
    """Full case record as returned by the API."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    alert_group_id: uuid.UUID
    title: str
    description: str | None
    severity: str
    risk_score: int
    status: str
    disposition: str | None
    created_at: datetime
    updated_at: datetime
    resolved_at: datetime | None
    closed_at: datetime | None


class IncidentListResponse(BaseModel):
    """GET /incidents — paged, newest first."""

    total: int
    page: int
    size: int
    items: list[IncidentRead]
